engine/src/test/java/com/redhat/cloud/notifications/routers/DailyDigestResourceTest.java

```diff
@@ -0,0 +1,143 @@
+package com.redhat.cloud.notifications.routers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.cloud.notifications.Json;
+import com.redhat.cloud.notifications.TestLifecycleManager;
+import com.redhat.cloud.notifications.models.AggregationCommand;
+import com.redhat.cloud.notifications.models.EmailAggregationKey;
+import com.redhat.cloud.notifications.models.EmailSubscriptionType;
+import com.redhat.cloud.notifications.routers.dailydigest.TriggerDailyDigestRequestDto;
+import com.redhat.cloud.notifications.templates.models.Environment;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+import io.restassured.http.ContentType;
+import io.smallrye.reactive.messaging.providers.connectors.InMemoryConnector;
+import io.smallrye.reactive.messaging.providers.connectors.InMemorySink;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import javax.enterprise.inject.Any;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.time.LocalDateTime;
+
+import static com.redhat.cloud.notifications.Constants.API_INTERNAL;
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+public class DailyDigestResourceTest {
+
+    @InjectMock
+    Environment environment;
+
+    @Any
+    @Inject
+    InMemoryConnector inMemoryConnector;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    /**
+     * Tests that the daily digest cannot be triggered in an environment that
+     * isn't "local" or "stage".
+     */
+    @Test
+    public void testTriggerDailyDigestNonStage() {
+        // Simulate that we are in the prod environment.
+        Mockito.when(this.environment.isEnvironmentLocal()).thenReturn(false);
+        Mockito.when(this.environment.isEnvironmentStage()).thenReturn(false);
+
+        final TriggerDailyDigestRequestDto triggerDailyDigestRequestDto = new TriggerDailyDigestRequestDto(
+            "application-name",
+            "bundle-name",
+            "organization-id",
+            null,
+            null
+        );
+
+        // Call the endpoint under test.
+        final String response = given()
+            .basePath(API_INTERNAL)
+            .when()
+            .contentType(JSON)
+            .body(Json.encode(triggerDailyDigestRequestDto))
+            .post("/daily-digest")
+            .then()
+            .statusCode(400)
+            .extract()
+            .asString();
+
+        Assertions.assertEquals("the daily digests can only be triggered in the stage environment", response, "unexpected error message received");
+    }
```

Comment:
For some reason in this test the `response` is empty and the assertion fails, but even though I've debugged the process and dig deep on this I am not able to see why it comes empty :thinking: :confused:

Reply:
Only this one is failing?

Reply:
Yes! I have tried launching the engine and hitting the endpoint and it also returns an empty body :confused:

Reply:
Perhaps a Produces(TEXT_PLAIN) ? 
If still not returning anything instead of returning void, it could return a `Response` and you can manually build the return value like its done [here](https://github.com/RedHatInsights/notifications-backend/blob/9fe178aa5bb690abe085a6dcf52a10c291a9457f/backend/src/main/java/com/redhat/cloud/notifications/routers/UserConfigResource.java#L99-L103).

Reply:
Perfect, I'll try that! Thank you very much :smile:

Reply:
Using `@Produces(TEXT_PLAIN)` had the same result... Although your suggestion to build the `Response` fixed the issue! Thank you :man_dancing: